plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/util/OrcMetrics.java

```diff
@@ -87,7 +87,7 @@ public static Metrics fileMetrics(TrinoInputFile file, MetricsConfig metricsConf
             Footer footer = reader.get().getFooter();
 
             // use name mapping to compute missing Iceberg field IDs
-            Optional<NameMapping> nameMapping = Optional.of(MappingUtil.create(schema));
+            NameMapping nameMapping = MappingUtil.create(schema);
```

Comment:
So if I'm understanding correctly, we have two different codepaths that eventually land at `fileColumnsByIcebergId`. One from `IcebergPageSourceProvider`, and one here in `OrcMetrics`. Prior to this PR, the latter would use `MappingUtil` to derive the schema, but the former would only use the table properties. And in this PR, we are standardizing so that both codepaths use `MappingUtil`. Do I have that right?

I'm curious, doesn't this `OrcMetrics` codepath also need to handle the case where the name mapping is set as a JSON in the table properties? Like why aren't these codepaths completely unified in how they derive a `NameMapping` from the combination of ORC files / table metadata / table schema?

Reply:
IIUC `OrcMetrics` are only used when migrating table to Iceberg, therefore the Iceberg table metadata is not yet created as well as the table properties. Column Ids are expected to be missing in the ORC files so we are using the name mapping derived from table schema to fill the gap.

Reply:
Makes sense, thanks for clarifying!